TestsGetAndFilterUsers.cs

```diff
@@ -0,0 +1,160 @@
+﻿using Newtonsoft.Json;
+using NUnit.Framework;
+using RestSharp;
+
+namespace APIAutomation.Tests
+{
+    [TestFixture]
+    public class GetAndFilterUsersTests
+    {
+        private RestClient _client;
+        private RestRequest _request;
+
+        [SetUp]
+        public void Setup()
+        {
+            var client = ClientForReadScope.GetInstance();
+            _client = client.GetRestClient();
+            _request = new RestRequest("/users");
+        }
+
+        [Test]
+        public void GetAllUsers_ReturnsAllExpectedUsers_Test()
+        {
+            _client.AddDefaultHeader("Accept", "application/json");
+            RestResponse response = _client.Execute(_request);
+
+            var expectedUsers = new List<User>
+            {
+                new User { Name = "Michael Davis", Age = 3, Sex = "MALE", ZipCode = "12345" },
+                new User { Name = "Olivia Johnson", Age = null, Sex = "FEMALE", ZipCode = null }
+            };
+
+            List<User> actualUsers = JsonConvert.DeserializeObject<List<User>>(response.Content);
+
+            Assert.Multiple(() =>
+            {
+                Assert.That((int)response.StatusCode, Is.EqualTo(200));
+                Assert.That(expectedUsers.Count, Is.EqualTo(actualUsers.Count), "Number of users in response doesn't match expected count!");
+                Assert.That(actualUsers.Distinct().Count(), Is.EqualTo(expectedUsers.Count), "Response contains duplicates for users.");
+
+                foreach (var actualUser in actualUsers)
```

Comment:
You may use here Assert.That(actualUsers, Is.EquivalentTo(expectedUsers), "Message"). In this case you will need to implement IEquatable interface for your User class and also override GetHashCode(). Here is a link with more details how to do that - https://stackoverflow.com/questions/10454519/best-way-to-compare-two-complex-objects. I would say that it's not necessary, but such implementation will be more effective:)

Reply:
Ok, it is done, but I'm not sure if it is needed to override also GetHashCode().